specification/app/resource-manager/Microsoft.App/preview/2023-08-01-preview/CommonDefinitions.json

```diff
@@ -1185,6 +1185,204 @@
+              "description": "Defines parameters for tcp connection pooling",
+              "$ref": "#/definitions/tcpConnectionPool"
+            }
+          }
+        }
+      }
+    },
+    "TimeoutPolicy": {
+      "description": "Policy to set request timeouts",
+      "type": "object",
+      "properties": {
+        "responseTimeoutInSeconds": {
+          "description": "Timeout, in seconds, for a request to respond",
+          "format": "int32",
+          "type": "integer"
+        },
+        "connectionTimeoutInSeconds": {
+          "description": "Timeout, in seconds, for a request to initiate a connection",
+          "format": "int32",
+          "type": "integer"
+        }
+      }
+    },
+    "HttpRetryPolicy": {
+      "description": "Policy that defines http request retry conditions",
+      "type": "object",
+      "properties": {
+        "maxRetries": {
+          "description": "Maximum number of times a request will retry",
+          "format": "int32",
+          "type": "integer"
+        },
+        "retryBackOff": {
+          "description": "Settings for retry backoff characteristics",
+          "type": "object",
+          "properties": {
+            "initialDelayInMilliseconds": {
+              "description": "Initial delay, in milliseconds, before retrying a request",
+              "format": "int64",
+              "type": "integer"
+            },
+            "maxIntervalInMilliseconds": {
+              "description": "Maximum interval, in milliseconds, between retries",
+              "format": "int64",
+              "type": "integer"
+            }
+          }
+        },
+        "matches": {
+          "description": "Conditions that must be met for a request to be retried",
+          "type": "object",
+          "properties": {
+            "headers": {
+              "description": "Headers that must be present for a request to be retried",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/HeaderMatch"
+              }
+            },
+            "httpStatusCodes": {
+              "description": "Additional http status codes that can trigger a retry",
+              "type": "array",
+              "items": {
+                "format": "int32",
+                "type": "integer"
+              }
+            },
+            "errors": {
+              "description": "Errors that can trigger a retry",
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            }
+          }
+        }
+      }
+    },
+    "TcpRetryPolicy": {
+      "description": "Policy that defines tcp request retry conditions",
+      "type": "object",
+      "properties": {
+        "maxConnectAttempts": {
+          "description": "Maximum number of attempts to connect to the tcp service",
+          "format": "int32",
+          "type": "integer"
+        }
+      }
+    },
+    "circuitBreakerPolicy": {
```

Comment:
We need to add a "max ejection percent" property to control the maximum number of upstream hosts that can be ejected. This will help prevent complete outages during transient failures. Envoy supports this setting in its outlier detection: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/outlier_detection.proto#envoy-v3-api-field-config-cluster-v3-outlierdetection-max-ejection-percent

Reply:
This comment probably belongs on the open RP and k4apps PRs instead @vturecek - but your comment makes sense.

Reply:
FYI - We have added this property now to data plane and RP API. But @halspang will need to update this PR here because we do not have write access to it.

Reply:
Added.